app/services/bulk_unsubscribe_service.rb

```diff
@@ -72,6 +74,29 @@ def self.send_email_to_subscriber(subscriber, subscription_details, index)
       queue: :delivery_immediate
     )
 
+    # Send a sample of the emails to the courtesy copy group. As
+    # we're iterating over subscribers here, sending each email
+    # would be excessive.
+    if (index % 500).zero?
```

Comment:
Instead of using 'index' here, maybe it is better to make this a boolean 'send_courtesy_copy?' and then compute that when calling this service. I think that would make the behaviour a bit clearer

Reply:
I did switch to that approach eventually, but I've got back and switched around the commits to avoid this intermediate state with the index, so hopefully the commit history is a bit neater now.